.github/workflows/release.yaml

```diff
@@ -2,6 +2,8 @@ name: "Release"
 
 on:
   push:
+    branches:
```

Comment:
This will also trigger the pypi releases. Could we separate this workflow into two workflows so we can be more explicit on the triggers?